model/src/cfd.rs

```diff
@@ -1036,26 +1036,16 @@ impl Cfd {
         }))
     }
 
-    pub fn reject_collaborative_settlement(self, reason: anyhow::Error) -> Result<CfdEvent> {
-        anyhow::ensure!(
-            self.can_settle_collaboratively(),
-            "Cannot reject collaborative settlement"
-        );
```

Comment:
❓ Do we still intend to eventually check the version here? Or does removing this mean that we want to be able to unconditionally emit these events?

Reply:
I think these should be unconditional for now. Version checks would be important if we want to make sure that "nothing else happened in between" (or similar). 
Given, that we have started an ephemeral protocol we always have to be able to fail/reject it (unconditionally), otherwise we might not allow other protocols (or even the same one) to start.

Reply:
I'd love to see some of these checks (like the one that `Completed/Failed` comes only after `Started`, but rather as `debug_assert!`s or just warnings when we rehydrate the Cfds - projection now deals gracefully with these events. 

Removing checks in this PR is orthogonal to that problem - I'm removing checks that are supposed to check for *starting* a protocol that are being re-purposed later on.